app-libs/stf/src/stf_sgx.rs

```diff
@@ -170,8 +178,8 @@ impl Stf {
 					)));
 					Ok(())
 				},
-				TrustedCall::balance_shield(root, who, value) => {
-					ensure_root(root)?;
+				TrustedCall::balance_shield(enclave_account, who, value) => {
+					ensure_self(enclave_account)?;
```

Comment:
Balance shield now ensures 'self' and not 'root', i.e. we make sure the account sender is the enclave account. The signature is verified upon insertion into the TOP pool.

Reply:
But we will leave this as a future issue: https://github.com/integritee-network/worker/issues/768

Reply:
Yes, good point, that will not work with this approach.

Reply:
Update: as we use the shielding key as basis for the enclave account, gossiping the transaction pool will be fine, I'll close #768